projects/go-lib/src/lib/components/go-datepicker/go-datepicker.component.ts

```diff
@@ -117,4 +117,8 @@ export class GoDatepickerComponent implements OnDestroy, OnInit {
   private initializePlaceholder(): void {
     this.placeholder = this.placeholder || LocaleFormat.format(this.locale);
   }
+  public reset(): void {
+    this.control.setValue(null);
+    this.selectedDate = this.control.value;
+  }
```

Comment:
I think we should opt to handle changes to the control from the implementation rather than doing this in the datepicker. All of our other form elements handle with the implementor calls `this.controlName.reset()` and we should do the same here. 

After digging into this, it seems like all we really _need_ to change here is handling when the new control value is `null` and set `selectedDate` accordingly.

In this snippet of code that exists in the `ngOnInit` method:
```ts
this.subscription = this.control.valueChanges.subscribe((value: Date) => {
  if (!value && this.selectedDate) {
    this.control.setErrors([{ message: 'format is invalid' }]);
  } else if (value) {
    this.selectedDate = LocaleFormat.formatDate(value, this.locale);
  }
});
```

We just need to change to:
```ts
this.subscription = this.control.valueChanges.subscribe((value: Date) => {
  if (!value) {
    this.selectedDate = null;
  } else {
    const dateValid: boolean = (value instanceof Date) && LocaleFormat.validDate(value.getMonth(), value.getDay(), value.getFullYear());
    if (!dateValid) {
      this.control.setErrors([{ message: 'format is invalid' }]);
    } else {
      this.selectedDate = LocaleFormat.formatDate(value, this.locale);
    }
  }
});
```

These changes will set the UI of the datepicker according to the change received on the control. If it's a falsy value, then it will null out the `selectedDate` and reset the UI to it's initial state. If its not a falsy value, we check if the date received is in and invalid format and set the errors if it is, if it's not then we set the `selectedDate` accordingly.

Reply:
Thank you for your suggestion

I think, If we will implement require validation in future then it will make problem in our application. Someone click on datepicker control and came out without insert any value it will throw the error. If we click on reset button then it will throw the same error 

This is also fine, We will change our code based on your comment.

Reply:
> Someone click on datepicker control and came out without insert any value it will throw the error

If the form element is required, this makes sense doesn't it?

I just want to make sure that we're picking the correct/best way for all of the applications moving forward.

Reply:
Yes, We have changed the code as per the above discussion and also updated the documentation in our application.